src/DynamoCoreWpf/ViewModels/Core/StateMachine.cs

```diff
@@ -520,7 +520,20 @@ internal bool HandleLeftButtonDown(object sender, MouseButtonEventArgs e)
                     {
                         if (e.ClickCount < 2) 
                         {
-                            InitiateWindowSelectionSequence();
+                            // if the Shift key is held down while the left mouse button is pressed, then do not initiate the window selection sequence
+
+                           if (Keyboard.Modifiers != ModifierKeys.Shift)
```

Comment:
1. Do positive condition checking if if, not negative condition checking. That is, `if (Keyboard.Modifiers == ModifierKeys.Shift)`
2. As `InitiateWindowSelectionSequence()` gets executed in both branches, it'd better to extract it out of if-else
3. Then if-else can be further simplified to `DynamoSelection.Instance.ClearSelectionDisabled = Keyboard.Modifier == ModifierKey.Shift;